nipype/algorithms/compcor.py

```diff
@@ -41,16 +43,16 @@ def _run_interface(self, runtime):
         X = (X - np.mean(X, axis=0)) / stdX
         u, _, _ = sp.linalg.svd(X, full_matrices=False)
         if components is None:
-            components = u[:, :num_components]
+            components = u[:, :self.inputs.num_components]
         else:
-            components = np.hstack((components, u[:, :num_components]))
+            components = np.hstack((components, u[:, :self.inputs.num_components]))
 
         components_file = os.path.join(os.getcwd(), 'noise_components.txt')
         np.savetxt(components_file, components, fmt="%.10f")
-        return components_file
+        return runtime
 
-class aCompCor(Node):
+class aCompCor(Workflow):
```

Comment:
Like @satra 's comment on the issue (https://github.com/nipy/nipype/issues/1594), I am proposing to write an interface containing the core CompCor logic. tCompCor and aCompCor would be workflows that contain nodes for creating the masks, a node for the core CompCor logic, and possibly (?) nodes for applying the extracted noise components. What do you think?